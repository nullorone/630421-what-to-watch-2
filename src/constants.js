export const EMPTY_STRING = ``;

export const Value = {
  EMPTY: 0,
  FULL: 1,
};

export const Pathname = {
  DEFAULT: `/`,
  DETAILS: `/details`,
  FILM: `/film`,
};

export const Img = {
  SMALL: {
    width: `280`,
    height: `175`
  },
  BIG: {
    width: `218`,
    height: `327`
  }
};

export const iconNames = [`add`, `fullscreen`, `in_list`, `pause`, `play`];

export const MAX_TIME_LENGTH = 3;

export const MOVIE_NAV_ITEMS = [
  {
    active: false,
    text: `Overview`,
  },
  {
    active: true,
    text: `Details`,
  },
  {
    active: false,
    text: `Reviews`,
  },
];

export const TypeCol = {
  TEXT: `text`,
  REVIEWS: `reviews`,
};

export const AmountSimilarFilms = {
  DEFAULT: 8,
  ON_PAGE_FILM: 4,
};

export const AMOUNT_ADDED_FILMS = 20;

export const MINUTES_IN_HOUR = 60;

export const SECONDS_IN_MINUTE = 60;

export const AMOUNT_COMMENT_IN_COL = 3;

export const MovieNavTabs = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

export const MILLISECONDS_IN_SECOND = 1000;

export const Url = {
  BASE: `https://htmlacademy-react-2.appspot.com/wtw`,
  FILMS: `/films`,
  PROMO: `/films/promo`,
  LOGIN: `/login`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`,
};

export const API_TIMEOUT = 5000;

export const Status = {
  AUTHORIZATION: 401,
  SUCCESS: 200,
};

export const MASK_AVATAR = `/static`;

export const Rating = {
  BAD: `bad`,
  NORMAL: `normal`,
  GOOD: `good`,
  VERY_GOOD: `very good`,
  AWESOME: `awesome`
};
