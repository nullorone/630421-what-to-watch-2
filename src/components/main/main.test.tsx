import * as React from 'react';
import renderer from "react-test-renderer";
import Main from './main';
import {EMPTY_STRING, Value} from "../../constants";
import {BrowserRouter} from "react-router-dom";

jest.mock(`../icons-wrapper/icons-wrapper`, () => `IconsWrapper`);
jest.mock(`../movie-card/movie-card`, () => `MovieCard`);
jest.mock(`../genre-list/genre-list`, () => `GenreList`);
jest.mock(`../movie-card-small-list/movie-card-small-list`, () => `MovieCardSmallList`);
jest.mock(`../logo/logo`, () => `Logo`);

describe(`Make snapshot`, () => {
  const initialProps = {
    films: [{
      id: Value.EMPTY,
      name: `Keks`,
      image: {
        poster: EMPTY_STRING,
        posterAlt: `1`,
        preview: EMPTY_STRING,
        previewAlt: EMPTY_STRING,
        background: EMPTY_STRING,
        backgroundAlt: EMPTY_STRING,
      },
      backgroundColor: EMPTY_STRING,
      video: {
        link: {
          mp4: EMPTY_STRING,
          webm: EMPTY_STRING,
        },
        poster: EMPTY_STRING,
      },
      description: EMPTY_STRING,
      rating: Value.FULL,
      scoresCount: Value.FULL,
      director: `Keks`,
      starring: [`Me`, `You`, `They`],
      runTime: Value.EMPTY,
      genre: EMPTY_STRING,
      released: Value.FULL,
      isFavorite: true,
    }],
    promo: {
      id: Value.EMPTY,
      name: `Keks`,
      image: {
        poster: EMPTY_STRING,
        posterAlt: `1`,
        preview: EMPTY_STRING,
        previewAlt: EMPTY_STRING,
        background: EMPTY_STRING,
        backgroundAlt: EMPTY_STRING,
      },
      backgroundColor: EMPTY_STRING,
      video: {
        link: EMPTY_STRING,
        preview: EMPTY_STRING,
      },
      description: EMPTY_STRING,
      rating: Value.FULL,
      scoresCount: Value.FULL,
      director: `Keks`,
      starring: [`Me`, `You`, `They`],
      runTime: Value.EMPTY,
      genre: EMPTY_STRING,
      released: Value.FULL,
      isFavorite: true,
    },
    selectedGenre: EMPTY_STRING,
    genres: [`ECMA`, `CSS`],
    icons: [`PAUSE`],
    onSelectedGenreClick: jest.fn(),
  };

  it(`Render main screen`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Main
              {...initialProps}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});


