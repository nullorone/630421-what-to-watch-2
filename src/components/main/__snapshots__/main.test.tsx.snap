// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make snapshot Render main screen 1`] = `
Array [
  <IconsWrapper
    iconNames={
      Array [
        "add",
        "fullscreen",
        "in_list",
        "pause",
        "play",
      ]
    }
  />,
  <MovieCard
    backgroundColor=""
    description=""
    director="Keks"
    genre=""
    id={0}
    image={
      Object {
        "background": "",
        "backgroundAlt": "",
        "poster": "",
        "posterAlt": "1",
        "preview": "",
        "previewAlt": "",
      }
    }
    isFavorite={true}
    name="Keks"
    onPlayButtonClick={[MockFunction]}
    rating={1}
    released={1}
    runTime={0}
    scoresCount={1}
    starring={
      Array [
        "Me",
        "You",
        "They",
      ]
    }
    video={
      Object {
        "link": Object {
          "mp4": "",
        },
        "poster": "",
      }
    }
  />,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <GenreList
        genres={
          Array [
            "ECMA",
            "CSS",
          ]
        }
        onGenreClick={[MockFunction]}
        selectedGenre=""
      />
    </section>
    <footer
      className="page-footer"
    >
      <Logo
        light={true}
      />
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
